File: VzBaseLibrary/src/main/java/com/eleven/baselibrary/utils/ActivityManager.java
package com.eleven.baselibrary.utils;

import android.app.Activity;

import java.lang.ref.WeakReference;
import java.util.Stack;

/**
 * Created by huangwz on 2017/6/3.
 *
 * @Link: vaitt_joy@163.com
 */

public class ActivityManager {
    public static final Stack<WeakReference<Activity>> activitys = new Stack<WeakReference<Activity>>();

    /**
     * 将Activity压入Application栈
     *
     * @param task 将要压入栈的Activity对象
     */
    public static void pushTask(WeakReference<Activity> task) {
        activitys.push(task);
    }

    /**
     * 将传入的Activity对象从栈中移除
     *
     * @param task
     */
    public static void removeTask(WeakReference<Activity> task) {
        activitys.remove(task);
    }

    /**
     * 根据指定位置从栈中移除Activity
     *
     * @param taskIndex Activity栈索引
     */
    public void removeTask(int taskIndex) {
        if (activitys.size() > taskIndex)
            activitys.remove(taskIndex);
    }

    /**
     * 将栈中Activity移除至栈顶
     */
    public void removeToTop() {
        int end = activitys.size();
        int start = 1;
        for (int i = end - 1; i >= start; i--) {
            if (!activitys.get(i).get().isFinishing()) {
                activitys.get(i).get().finish();
            }
        }
    }

    /**
     * 移除全部（用于整个应用退出）
     */
    public void removeAll() {
        //finish所有的Activity
        for (WeakReference<Activity> task : activitys) {
            if (!task.get().isFinishing()) {
                task.get().finish();
            }
        }
    }
}

File: app/src/main/java/com/vz/phpkotlin/utils/ActivityManager.kt
package com.vz.phpkotlin.utils

import android.app.Activity
import java.lang.ref.WeakReference
import java.util.*

/**
 * Created by huangwz on 2017/4/25.
 * app activity 管理
 */

class ActivityManager {

    /**
     * 根据指定位置从栈中移除Activity

     * @param taskIndex Activity栈索引
     */
    fun removeTask(taskIndex: Int) {
        if (activitys.size > taskIndex)
            activitys.removeAt(taskIndex)
    }

    /**
     * 将栈中Activity移除至栈顶
     */
    fun removeToTop() {
        val end = activitys.size
        val start = 1
        for (i in end - 1 downTo start) {
            if (!activitys[i].get()!!.isFinishing()) {
                activitys[i].get()!!.finish()
            }
        }
    }

    /**
     * 移除全部（用于整个应用退出）
     */
    fun removeAll() {
        //finish所有的Activity
        for (task in activitys) {
            if (!task.get()!!.isFinishing()) {
                task.get()!!.finish()
            }
        }
    }

    companion object {

        val activitys = Stack<WeakReference<Activity>>()

        /**
         * 将Activity压入Application栈

         * @param task 将要压入栈的Activity对象
         */
        fun pushTask(task: WeakReference<Activity>) {
            activitys.push(task)
        }

        /**
         * 将传入的Activity对象从栈中移除

         * @param task
         */
        fun removeTask(task: WeakReference<Activity>) {
            activitys.remove(task)
        }
    }
}
